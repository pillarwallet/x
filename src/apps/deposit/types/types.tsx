export type BalanceInfo = {
  type?: 'BalanceInfo';
  name: string;
  chain: string;
  address: string;
  decimals: number;
  symbol: string;
  logoURI: string;
  balance: string;
};

export type AddedAssets = {
  type?: 'AddedAsset';
  assetType: 'token' | 'nft';
  tokenId?: string;
  chain: string;
  tokenAddress: string;
  balance: number;
};

export type Network =
  | 'ethereum'
  | 'polygon'
  | 'gnosis'
  | 'base'
  | 'bnb smart chain'
  | 'optimism'
  | 'arbitrum';

export type TokenList = {
  address: string;
  chainId: number;
  decimals: string;
  name: string;
  symbol: string;
  logoURI: string;
};

export type BalancesByChain = {
  [chain: string]: BalanceInfo[];
};
