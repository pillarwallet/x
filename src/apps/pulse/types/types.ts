export type TransactionStatusState =
  | 'Starting Transaction'
  | 'Transaction Pending'
  | 'Transaction Complete'
  | 'Transaction Failed';

export type TransactionStep =
  | 'Submitted'
  | 'Pending'
  | 'ResourceLock'
  | 'Completed';

export type StepStatus = 'completed' | 'pending' | 'failed' | 'inactive';

export interface TokenInfo {
  symbol: string;
  name: string;
  logo: string;
  address?: string;
}

export interface SellOffer {
  tokenAmountToReceive: number;
  minimumReceive: number;
}

export interface PayingToken {
  totalUsd: number;
  name: string;
  symbol: string;
  logo: string;
  actualBal: string;
  totalRaw: string;
  chainId: number;
  address: string;
}

export interface TokenDetails {
  symbol: string;
  name: string;
  address: string;
  chainId: number;
  amount: string;
  logo: string;
  type: 'BUY_TOKEN' | 'SELL_TOKEN';
}

export interface BuyModeDetails {
  usdAmount: string;
  payingTokens: PayingToken[];
  totalPayingUsd: number;
}

export interface TransactionStatusConfig {
  icon: string;
  containerClasses: string;
  iconClasses: string;
  color: string;
}

export interface ButtonConfig {
  bgColor: string;
  textColor: string;
  borderColor: string;
  label: string;
}

export interface ProgressStepConfig {
  label: string;
  order: number;
}

export interface TransactionStatusProps {
  closeTransactionStatus: () => void;
  userOpHash: string;
  chainId: number;
  gasFee?: string;
  // Transaction data from PreviewSell/PreviewBuy
  isBuy?: boolean;
  sellToken?: TokenInfo | null;
  buyToken?: TokenInfo | null;
  tokenAmount?: string;
  sellOffer?: SellOffer | null;
  payingTokens?: PayingToken[];
  usdAmount?: string;
  // Externalized polling state
  currentStatus: TransactionStatusState;
  errorDetails: string;
  submittedAt?: Date;
  pendingCompletedAt?: Date;
  blockchainTxHash?: string;
  resourceLockTxHash?: string;
  completedTxHash?: string;
  completedChainId?: number;
  resourceLockChainId?: number;
  resourceLockCompletedAt?: Date;
  isResourceLockFailed?: boolean;
}

export interface TransactionDetailsProps {
  onDone: () => void;
  userOpHash: string;
  chainId: number;
  status: TransactionStatusState;
  // Transaction data from PreviewSell/PreviewBuy
  isBuy?: boolean;
  sellToken?: TokenInfo | null;
  buyToken?: TokenInfo | null;
  tokenAmount?: string;
  sellOffer?: SellOffer | null;
  payingTokens?: PayingToken[];
  usdAmount?: string;
  submittedAt?: Date;
  pendingCompletedAt?: Date;
  resourceLockCompletedAt?: Date;
  txHash?: string;
  gasFee?: string;
  errorDetails?: string;
  resourceLockTxHash?: string;
  completedTxHash?: string;
  resourceLockChainId?: number;
  completedChainId?: number;
  isResourceLockFailed?: boolean;
}

export interface TransactionInfoProps {
  status: TransactionStatusState;
  userOpHash: string;
  txHash?: string;
  chainId: number;
  gasFee?: string;
  completedAt?: Date;
  // Buy-specific fields
  isBuy?: boolean;
  resourceLockTxHash?: string;
  resourceLockChainId?: number;
  completedTxHash?: string;
  completedChainId?: number;
}

export interface ProgressStepProps {
  step: TransactionStep;
  status: StepStatus;
  label: string;
  isLast?: boolean;
  showLine?: boolean;
  lineStatus?: StepStatus;
  timestamp?: string | React.ReactNode;
}

export interface TransactionErrorBoxProps {
  technicalDetails?: string;
}

export interface UseClickOutsideOptions {
  ref: React.RefObject<HTMLElement>;
  callback: () => void;
  condition: boolean;
}

export interface UseKeyboardNavigationOptions {
  onEscape: () => void;
  onEnter?: () => void;
  enabled?: boolean;
}

export interface TechnicalDetails {
  transactionType: 'BUY' | 'SELL';
  transactionHash: string;
  hashType: 'bidHash' | 'userOpHash';
  chainId: number;
  status: TransactionStatusState;
  timestamp: string;
  accountAddress: string;
  token: TokenDetails | null;
  sellOffer: SellOffer | null;
  buyMode: BuyModeDetails | null;
  transactionHashes: {
    [key: string]: string;
  };
  chains: {
    mainChainId: number;
    resourceLockChainId: string | number;
    completedChainId: string | number;
  };
  timestamps: {
    [key: string]: string;
  };
  error: {
    details: string;
    isResourceLockFailed: boolean;
    failureStep: string;
  };
  gas: {
    fee: string;
  };
  stepStatus: {
    [key: string]: StepStatus;
  };
}
