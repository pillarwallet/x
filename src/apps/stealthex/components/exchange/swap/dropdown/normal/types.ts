
import type React from 'react';
import { StructuredCurrency } from '../../../../../type';
import { CurrencyData } from '../../../../../lib/backend/api';

export type ClickCallback = (currency: CurrencyData) => void;

export type AvailableCurrency =
  | {
    availableCurrency?: Omit<StructuredCurrency, 'fiat'> & {
      all?: CurrencyData[];
    };
    type: 'get';
  }
  | {
    availableCurrency?: StructuredCurrency & {
      all?: CurrencyData[];
    };
    type: 'send';
  };

export type NormalDropdownProps = AvailableCurrency & {
  loading?: boolean;
  selectedCurrency: CurrencyData | null;
  searchContext: string;
  fiat?: boolean;
  onClick?: ClickCallback;
  components?: {
    NotFound?: React.FC<React.PropsWithChildren>;
    Group?: React.FC<React.PropsWithChildren>;
    Item?: React.FC<
      React.PropsWithChildren<
        | { loading: true }
        | {
          loading?: false;
          currency: CurrencyData;
          selected?: boolean;
          onClick?: () => void;
        }
      >
    >;
  };
  groupHeight?: number;
  itemHeight?: number;
  maxHeight?: number;
  embed?: boolean;
};
