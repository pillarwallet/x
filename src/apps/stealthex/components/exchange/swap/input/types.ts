
import { CurrencyData } from '../../../../lib/backend/api';
import type { AvailableCurrency } from '../dropdown/normal/types';

export type SavedAvailableCurrency =
  | {
      floatingDropdown: true;
      availableCurrency: CurrencyData[];
      type: 'get' | 'send';
    }
  | ({
      floatingDropdown?: false;
    } & AvailableCurrency);

export type InputProps = {
  currency: CurrencyData;
  value?: string;
  loading?: boolean;
  loadingCurrency?: boolean;
  error?: string | JSX.Element;
  fiat?: boolean;
  disabled?: boolean;
  withLock?: boolean | null;
  widget?: boolean;
  fixed?: boolean;
  onChange?: React.ChangeEventHandler<HTMLInputElement>;
  onFixedChange?: (fixed: boolean) => void;
  onCurrencySelect?: (currency: CurrencyData) => void;
  pressable?: boolean;
} & SavedAvailableCurrency;
