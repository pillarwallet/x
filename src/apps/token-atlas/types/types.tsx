export type ChainType = {
  chainId: number;
  chainName: string;
};

export enum PeriodFilter {
  HOUR = 'HOUR',
  DAY = 'DAY',
  WEEK = 'WEEK',
  MONTH = 'MONTH',
  YEAR = 'YEAR',
}

export type SelectedTokenType = {
  id: number;
  symbol: string;
  address: string;
  decimals?: number;
  chainId?: number;
  name: string;
  icon?: string;
  price?: number;
};
