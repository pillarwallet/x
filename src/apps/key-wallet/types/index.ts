export interface Asset {
  id: number;
  name: string;
  symbol: string;
  logo: string;
  balance: number;
  decimals: number;
  price: number;
  price_change_24h: number;
  contract: string;
  chainId: number;
  chainName: string;
  usdBalance: number;
}

export interface SendFormData {
  amount: string;
  recipient: string;
}

export interface TransactionStatus {
  hash: string;
  chainId: number;
  status: 'pending' | 'success' | 'failed';
  timestamp: number;
}

