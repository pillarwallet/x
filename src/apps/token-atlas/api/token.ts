import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';

// store
import { addMiddleware } from '../../../store';

// types
import {
  BlockchainList,
  TokenAtlasGraphApiResponse,
  TokenAtlasInfoApiResponse,
  TrendingTokens,
} from '../../../types/api';

// utils
import { CompatibleChains } from '../../../utils/blockchain';

const chainIds =
  process.env.REACT_APP_USE_TESTNETS === 'true'
    ? [11155111]
    : CompatibleChains.map((chain) => chain.chainId);
const chainIdsQuery = chainIds.map((id) => `chainIds=${id}`).join('&');

export const tokenInfoApi = createApi({
  reducerPath: 'tokenInfoApi',
  baseQuery: fetchBaseQuery({
    baseUrl:
      process.env.REACT_APP_USE_TESTNETS === 'true'
        ? 'https://token-nubpgwxpiq-uc.a.run.app'
        : 'https://token-7eu4izffpa-uc.a.run.app',
  }),
  endpoints: (builder) => ({
    getTokenInfo: builder.query<
      TokenAtlasInfoApiResponse,
      { asset: string; blockchain: string; symbol: string }
    >({
      query: ({ asset, blockchain, symbol }) => {
        const blockchainParam =
          blockchain !== undefined ? `&blockchain=${blockchain}` : '';
        return `?asset=${asset}&symbol=${symbol}${blockchainParam}&${chainIdsQuery}&testnets=${process.env.REACT_APP_USE_TESTNETS || 'true'}`;
      },
    }),
  }),
});

export const tokenGraphApi = createApi({
  reducerPath: 'tokenGraphApi',
  baseQuery: fetchBaseQuery({
    baseUrl:
      process.env.REACT_APP_USE_TESTNETS === 'true'
        ? 'https://tokenpricehistory-nubpgwxpiq-uc.a.run.app'
        : 'https://tokenpricehistory-7eu4izffpa-uc.a.run.app',
  }),
  endpoints: (builder) => ({
    getTokenGraph: builder.query<
      TokenAtlasGraphApiResponse,
      { blockchain?: string; asset: string; from: number; to?: number }
    >({
      query: ({ blockchain, asset, from, to }) => {
        const toParam = to !== undefined ? `&to=${from * 1000}` : '';
        const blockchainParam =
          blockchain !== undefined ? `&blockchain=${blockchain}` : '';
        const assetParam = asset.split(' ')[0];
        return `?asset=${assetParam}&from=${from * 1000}${toParam}${blockchainParam}&${chainIdsQuery}&testnets=${process.env.REACT_APP_USE_TESTNETS || 'true'}`;
      },
    }),
  }),
});

export const trendingTokensApi = createApi({
  reducerPath: 'trendingTokensApi',
  baseQuery: fetchBaseQuery({
    baseUrl:
      process.env.REACT_APP_USE_TESTNETS === 'true'
        ? 'https://trendingtokens-nubpgwxpiq-uc.a.run.app'
        : 'https://trendingtokens-7eu4izffpa-uc.a.run.app',
  }),
  endpoints: (builder) => ({
    getTrendingTokens: builder.query<TrendingTokens, void>({
      query: () =>
        `?${chainIdsQuery}&testnets=${process.env.REACT_APP_USE_TESTNETS || 'true'}`,
    }),
  }),
});

export const blockchainsListApi = createApi({
  reducerPath: 'blockchainsListApi',
  baseQuery: fetchBaseQuery({
    baseUrl:
      process.env.REACT_APP_USE_TESTNETS === 'true'
        ? 'https://blockchains-nubpgwxpiq-uc.a.run.app'
        : 'https://blockchains-7eu4izffpa-uc.a.run.app',
  }),
  endpoints: (builder) => ({
    getBlockchainsList: builder.query<BlockchainList, void>({
      query: () =>
        `?${chainIdsQuery}&testnets=${process.env.REACT_APP_USE_TESTNETS || 'true'}`,
    }),
  }),
});

/**
 * Add this to the store
 */
addMiddleware(tokenInfoApi);
addMiddleware(tokenGraphApi);
addMiddleware(trendingTokensApi);
addMiddleware(blockchainsListApi);

export const { useGetTokenInfoQuery } = tokenInfoApi;
export const { useGetTokenGraphQuery } = tokenGraphApi;
export const { useGetTrendingTokensQuery } = trendingTokensApi;
export const { useGetBlockchainsListQuery } = blockchainsListApi;
