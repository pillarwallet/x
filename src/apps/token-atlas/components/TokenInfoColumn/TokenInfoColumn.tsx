import { CircularProgress } from '@mui/material';
import { createRef } from 'react';

// hooks
import useRefDimensions from '../../../pillarx-app/hooks/useRefDimensions';
import { useAppDispatch, useAppSelector } from '../../hooks/useReducerHooks';

// reducer
import { setIsAllChainsVisible } from '../../reducer/tokenAtlasSlice';

// types
import { TokenAtlasInfoData } from '../../../../types/api';

// components
import ChainCard from '../ChainCard/ChainCard';
import PriceCard from '../PriceCard/PriceCard';
import Body from '../Typography/Body';
import BodyLight from '../Typography/BodyLight';

type TokenInfoColumnProps = {
  isLoadingTokenDataInfo: boolean;
  className?: string;
};

const TokenInfoColumn = ({
  className,
  isLoadingTokenDataInfo,
}: TokenInfoColumnProps) => {
  const dispatch = useAppDispatch();
  const tokenDataInfo = useAppSelector(
    (state) => state.tokenAtlas.tokenDataInfo as TokenAtlasInfoData | undefined
  );
  const isAllChainsVisible = useAppSelector(
    (state) => state.tokenAtlas.isAllChainsVisible as boolean
  );
  const divRef = createRef<HTMLDivElement>();
  const dimensions = useRefDimensions(divRef);

  // This is to make sure that if the chains do not all fit in one line, we add the number of hidden chains
  const tokenBlockchainsList =
    tokenDataInfo?.contracts.map((contract) => contract.blockchain) || [];
  const numberVisibleCards = Math.floor((dimensions.width - 50) / 158);
  const numberHiddenCards = Math.max(
    tokenBlockchainsList.length - numberVisibleCards,
    0
  );

  const handleShowAllChains = () => {
    dispatch(setIsAllChainsVisible(!isAllChainsVisible));
  };

  const priceChanges = [
    { period: '1H', percentage: tokenDataInfo?.price_change_1h },
    { period: '24H', percentage: tokenDataInfo?.price_change_24h },
    { period: '7D', percentage: tokenDataInfo?.price_change_7d },
    { period: '1M', percentage: tokenDataInfo?.price_change_1m },
    { period: '1Y', percentage: tokenDataInfo?.price_change_1y },
  ];

  return (
    <div
      id="token-atlas-token-info-column"
      ref={divRef}
      className={`flex flex-col gap-10 ${className}`}
    >
      <div className="flex flex-col gap-2">
        <Body>Blockchains</Body>
        <div
          id="token-atlas-info-column-blockchain-list"
          className="w-full h-fit flex flex-wrap gap-2"
        >
          {isLoadingTokenDataInfo && (
            <CircularProgress
              size={32}
              sx={{ color: '#979797' }}
              data-testid="circular-loading"
            />
          )}
          {isAllChainsVisible
            ? tokenBlockchainsList.map((chain, index) => (
                <ChainCard key={index} chainName={chain} />
              ))
            : tokenBlockchainsList
                .slice(0, numberVisibleCards)
                .map((chain, index) => (
                  <ChainCard key={index} chainName={chain} />
                ))}
          {numberHiddenCards !== 0 && (
            <div
              className="flex rounded-full bg-medium_grey p-2 items-center h-8 cursor-pointer"
              onClick={handleShowAllChains}
            >
              {numberHiddenCards > 0 && !isAllChainsVisible ? (
                <Body>+ {numberHiddenCards}</Body>
              ) : (
                <Body className="text-[9px] font-medium">Show less</Body>
              )}
            </div>
          )}
        </div>
      </div>
      <div
        id="token-atlas-info-column-price-change-cards"
        className="flex flex-col gap-2"
      >
        <Body>Price changes</Body>
        <div className="flex flex-wrap gap-2">
          {isLoadingTokenDataInfo && (
            <CircularProgress
              size={32}
              sx={{ color: '#979797' }}
              data-testid="circular-loading"
            />
          )}
          {priceChanges.map((price, index) => (
            <PriceCard
              key={index}
              percentage={price.percentage}
              timePeriod={price.period}
            />
          ))}
        </div>
      </div>
      <div
        id="token-atlas-info-column-stats"
        className="flex flex-col gap-2 mb-20"
      >
        <Body>Stats</Body>
        <div className="flex flex-col rounded px-4 bg-medium_grey">
          <div className="flex justify-between border-b border-b-dark_grey py-3">
            <BodyLight>All time high</BodyLight>
            <Body>{tokenDataInfo?.ath || '-'}</Body>
          </div>
          <div className="flex justify-between border-b border-b-dark_grey py-3">
            <BodyLight>All time low</BodyLight>
            <Body>{tokenDataInfo?.atl || '-'}</Body>
          </div>
          <div className="flex justify-between border-b border-b-dark_grey py-3">
            <BodyLight>Total supply</BodyLight>
            <Body>{tokenDataInfo?.total_supply || '-'}</Body>
          </div>
          <div className="flex justify-between border-b border-b-dark_grey py-3">
            <BodyLight>Circulating supply</BodyLight>
            <Body>{tokenDataInfo?.circulating_supply || '-'}</Body>
          </div>
          <div className="flex justify-between border-b border-b-dark_grey py-3">
            <BodyLight>Diluted market cap</BodyLight>
            <Body>{tokenDataInfo?.market_cap_diluted || '-'}</Body>
          </div>
        </div>
      </div>
    </div>
  );
};

export default TokenInfoColumn;
