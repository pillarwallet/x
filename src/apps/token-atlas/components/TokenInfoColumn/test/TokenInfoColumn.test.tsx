import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

// redux store
import { Provider } from 'react-redux';
import { store } from '../../../../../store';

// components
import TokenInfoColumn from '../TokenInfoColumn';

// reducer
import {
  setIsAllChainsVisible,
  setTokenDataInfo,
} from '../../../reducer/tokenAtlasSlice';

// types
import { TokenAtlasInfoData } from '../../../../../types/api';

const mockTokenDataInfo: TokenAtlasInfoData = {
  id: 1,
  market_cap: 100,
  market_cap_diluted: 100,
  liquidity: 150,
  price: 105,
  off_chain_volume: 150,
  volume: 150,
  volume_change_24h: 1.54,
  volume_7d: 2.3,
  is_listed: true,
  price_change_24h: 0.44,
  price_change_1h: 0.1,
  price_change_7d: 1.8,
  price_change_1m: 3.4,
  price_change_1y: 6.7,
  ath: 146,
  atl: 96,
  name: 'TOKEN',
  symbol: 'TKN',
  logo: 'tokenLogo.png',
  rank: 1047,
  contracts: [
    { blockchain: 'Ethereum', address: '0x01' },
    { blockchain: 'Binance Smart Chain', address: '0x02' },
    { blockchain: 'Polygon', address: '0x03' },
    { blockchain: 'Avalanche', address: '0x04' },
  ],
  total_supply: '300',
  circulating_supply: '170',
};

describe('<TokenInfoColumn />', () => {
  beforeEach(() => {
    store.dispatch(setIsAllChainsVisible(false));
    store.dispatch(setTokenDataInfo(mockTokenDataInfo));
  });

  it('renders correctly and matches snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <TokenInfoColumn isLoadingTokenDataInfo={false} />
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('displays circular loading when loading data', () => {
    render(
      <Provider store={store}>
        <TokenInfoColumn isLoadingTokenDataInfo />
      </Provider>
    );

    expect(screen.getAllByTestId('circular-loading')).toHaveLength(2);
  });

  it('renders some of the blockchain cards when not loading', () => {
    render(
      <Provider store={store}>
        <TokenInfoColumn isLoadingTokenDataInfo={false} />
      </Provider>
    );

    expect(screen.getByText('Ethereum')).toBeInTheDocument();
  });

  it('renders the correct price change cards', () => {
    render(
      <Provider store={store}>
        <TokenInfoColumn isLoadingTokenDataInfo={false} />
      </Provider>
    );

    expect(screen.getByText('1H')).toBeInTheDocument();
    expect(screen.getByText('24H')).toBeInTheDocument();
    expect(screen.getByText('7D')).toBeInTheDocument();
    expect(screen.getByText('1M')).toBeInTheDocument();
    expect(screen.getByText('1Y')).toBeInTheDocument();
  });

  it('renders correct stats for token information', () => {
    render(
      <Provider store={store}>
        <TokenInfoColumn isLoadingTokenDataInfo={false} />
      </Provider>
    );

    expect(screen.getByText('All time high')).toBeInTheDocument();
    expect(screen.getByText('146')).toBeInTheDocument();
    expect(screen.getByText('All time low')).toBeInTheDocument();
    expect(screen.getByText('96')).toBeInTheDocument();
    expect(screen.getByText('Total supply')).toBeInTheDocument();
    expect(screen.getByText('300')).toBeInTheDocument();
    expect(screen.getByText('Circulating supply')).toBeInTheDocument();
    expect(screen.getByText('170')).toBeInTheDocument();
    expect(screen.getByText('Diluted market cap')).toBeInTheDocument();
    expect(screen.getByText('100')).toBeInTheDocument();
  });

  it('does not crash when tokenDataInfo is undefined', () => {
    render(
      <Provider store={store}>
        <TokenInfoColumn isLoadingTokenDataInfo={false} />
      </Provider>
    );

    expect(screen.getByText('Blockchains')).toBeInTheDocument();
    expect(screen.getByText('Price changes')).toBeInTheDocument();
    expect(screen.getByText('Stats')).toBeInTheDocument();
  });
});
