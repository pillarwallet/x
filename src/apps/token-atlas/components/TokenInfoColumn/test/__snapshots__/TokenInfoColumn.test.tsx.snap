// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TokenInfoColumn /> renders correctly and matches snapshot 1`] = `
<div
  className="flex flex-col gap-10 undefined"
  id="token-atlas-token-info-column"
>
  <div
    className="flex flex-col gap-2"
  >
    <p
      className="text-[13px] font-semibold undefined"
    >
      Blockchains
    </p>
    <div
      className="w-full h-fit flex flex-wrap gap-2"
      id="token-atlas-info-column-blockchain-list"
    >
      <a
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          className="flex rounded-[50px] bg-medium_grey p-1 pr-3 items-center h-8 max-w-[150px] undefined"
          id="token-atlas-chain-card"
        >
          <div
            className="w-[24px] h-[24px] object-fill rounded-full mr-2 overflow-hidden"
          >
            <svg
              className="rounded-md"
              data-testid="random-avatar"
              fill="none"
              role="img"
              viewBox="0 0 80 80"
              xmlns="http://www.w3.org/2000/svg"
            >
              <mask
                height={80}
                id=":r0:"
                maskUnits="userSpaceOnUse"
                width={80}
                x={0}
                y={0}
              >
                <rect
                  fill="#FFFFFF"
                  height={80}
                  width={80}
                />
              </mask>
              <g
                mask="url(#:r0:)"
              >
                <rect
                  fill="#146A7C"
                  height={80}
                  width={80}
                />
                <path
                  d="M32.414 59.35L50.376 70.5H72.5v-71H33.728L26.5 13.381l19.057 27.08L32.414 59.35z"
                  fill="#F0AB3D"
                  filter="url(#filter_:r0:)"
                  transform="translate(-2 2) rotate(-322 40 40) scale(1.5)"
                />
                <path
                  d="M22.216 24L0 46.75l14.108 38.129L78 86l-3.081-59.276-22.378 4.005 12.972 20.186-23.35 27.395L22.215 24z"
                  fill="#C271B4"
                  filter="url(#filter_:r0:)"
                  style={
                    {
                      "mixBlendMode": "overlay",
                    }
                  }
                  transform="translate(-3 -3) rotate(-123 40 40) scale(1.5)"
                />
              </g>
              <defs>
                <filter
                  colorInterpolationFilters="sRGB"
                  filterUnits="userSpaceOnUse"
                  id="filter_:r0:"
                >
                  <feFlood
                    floodOpacity={0}
                    result="BackgroundImageFix"
                  />
                  <feBlend
                    in="SourceGraphic"
                    in2="BackgroundImageFix"
                    result="shape"
                  />
                  <feGaussianBlur
                    result="effect1_foregroundBlur"
                    stdDeviation={7}
                  />
                </filter>
              </defs>
            </svg>
          </div>
          <p
            className="text-[13px] font-semibold truncate capitalize"
          >
            Ethereum
          </p>
        </div>
      </a>
      <a
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          className="flex rounded-[50px] bg-medium_grey p-1 pr-3 items-center h-8 max-w-[150px] undefined"
          id="token-atlas-chain-card"
        >
          <div
            className="w-[24px] h-[24px] object-fill rounded-full mr-2 overflow-hidden"
          >
            <svg
              className="rounded-md"
              data-testid="random-avatar"
              fill="none"
              role="img"
              viewBox="0 0 80 80"
              xmlns="http://www.w3.org/2000/svg"
            >
              <mask
                height={80}
                id=":r1:"
                maskUnits="userSpaceOnUse"
                width={80}
                x={0}
                y={0}
              >
                <rect
                  fill="#FFFFFF"
                  height={80}
                  width={80}
                />
              </mask>
              <g
                mask="url(#:r1:)"
              >
                <rect
                  fill="#C271B4"
                  height={80}
                  width={80}
                />
                <path
                  d="M32.414 59.35L50.376 70.5H72.5v-71H33.728L26.5 13.381l19.057 27.08L32.414 59.35z"
                  fill="#C20D90"
                  filter="url(#filter_:r1:)"
                  transform="translate(0 0) rotate(16 40 40) scale(1.2)"
                />
                <path
                  d="M22.216 24L0 46.75l14.108 38.129L78 86l-3.081-59.276-22.378 4.005 12.972 20.186-23.35 27.395L22.215 24z"
                  fill="#92A1C6"
                  filter="url(#filter_:r1:)"
                  style={
                    {
                      "mixBlendMode": "overlay",
                    }
                  }
                  transform="translate(0 0) rotate(-24 40 40) scale(1.2)"
                />
              </g>
              <defs>
                <filter
                  colorInterpolationFilters="sRGB"
                  filterUnits="userSpaceOnUse"
                  id="filter_:r1:"
                >
                  <feFlood
                    floodOpacity={0}
                    result="BackgroundImageFix"
                  />
                  <feBlend
                    in="SourceGraphic"
                    in2="BackgroundImageFix"
                    result="shape"
                  />
                  <feGaussianBlur
                    result="effect1_foregroundBlur"
                    stdDeviation={7}
                  />
                </filter>
              </defs>
            </svg>
          </div>
          <p
            className="text-[13px] font-semibold truncate capitalize"
          >
            Binance Smart Chain
          </p>
        </div>
      </a>
      <a
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          className="flex rounded-[50px] bg-medium_grey p-1 pr-3 items-center h-8 max-w-[150px] undefined"
          id="token-atlas-chain-card"
        >
          <div
            className="w-[24px] h-[24px] object-fill rounded-full mr-2 overflow-hidden"
          >
            <svg
              className="rounded-md"
              data-testid="random-avatar"
              fill="none"
              role="img"
              viewBox="0 0 80 80"
              xmlns="http://www.w3.org/2000/svg"
            >
              <mask
                height={80}
                id=":r2:"
                maskUnits="userSpaceOnUse"
                width={80}
                x={0}
                y={0}
              >
                <rect
                  fill="#FFFFFF"
                  height={80}
                  width={80}
                />
              </mask>
              <g
                mask="url(#:r2:)"
              >
                <rect
                  fill="#F0AB3D"
                  height={80}
                  width={80}
                />
                <path
                  d="M32.414 59.35L50.376 70.5H72.5v-71H33.728L26.5 13.381l19.057 27.08L32.414 59.35z"
                  fill="#C271B4"
                  filter="url(#filter_:r2:)"
                  transform="translate(-4 -4) rotate(-284 40 40) scale(1.4)"
                />
                <path
                  d="M22.216 24L0 46.75l14.108 38.129L78 86l-3.081-59.276-22.378 4.005 12.972 20.186-23.35 27.395L22.215 24z"
                  fill="#C20D90"
                  filter="url(#filter_:r2:)"
                  style={
                    {
                      "mixBlendMode": "overlay",
                    }
                  }
                  transform="translate(-6 6) rotate(-246 40 40) scale(1.4)"
                />
              </g>
              <defs>
                <filter
                  colorInterpolationFilters="sRGB"
                  filterUnits="userSpaceOnUse"
                  id="filter_:r2:"
                >
                  <feFlood
                    floodOpacity={0}
                    result="BackgroundImageFix"
                  />
                  <feBlend
                    in="SourceGraphic"
                    in2="BackgroundImageFix"
                    result="shape"
                  />
                  <feGaussianBlur
                    result="effect1_foregroundBlur"
                    stdDeviation={7}
                  />
                </filter>
              </defs>
            </svg>
          </div>
          <p
            className="text-[13px] font-semibold truncate capitalize"
          >
            Polygon
          </p>
        </div>
      </a>
      <div
        className="flex rounded-full bg-medium_grey p-2 items-center h-8 cursor-pointer"
        onClick={[Function]}
      >
        <p
          className="text-[13px] font-semibold undefined"
        >
          + 
          5
        </p>
      </div>
    </div>
  </div>
  <div
    className="flex flex-col gap-2"
    id="token-atlas-info-column-price-change-cards"
  >
    <p
      className="text-[13px] font-semibold undefined"
    >
      Price changes
    </p>
    <div
      className="flex flex-wrap gap-2"
    >
      <div
        className="flex flex-col bg-medium_grey rounded p-2 gap-1 w-[125px]"
        id="token-atlas-price-card"
      >
        <p
          className="text-[9px] font-semibold text-light_grey"
        >
          1H
        </p>
        <p
          className="text-[13px] font-semibold undefined"
        >
          0.1
          %
        </p>
      </div>
      <div
        className="flex flex-col bg-medium_grey rounded p-2 gap-1 w-[125px]"
        id="token-atlas-price-card"
      >
        <p
          className="text-[9px] font-semibold text-light_grey"
        >
          24H
        </p>
        <p
          className="text-[13px] font-semibold undefined"
        >
          0.44
          %
        </p>
      </div>
      <div
        className="flex flex-col bg-medium_grey rounded p-2 gap-1 w-[125px]"
        id="token-atlas-price-card"
      >
        <p
          className="text-[9px] font-semibold text-light_grey"
        >
          7D
        </p>
        <p
          className="text-[13px] font-semibold undefined"
        >
          1.8
          %
        </p>
      </div>
      <div
        className="flex flex-col bg-medium_grey rounded p-2 gap-1 w-[125px]"
        id="token-atlas-price-card"
      >
        <p
          className="text-[9px] font-semibold text-light_grey"
        >
          1M
        </p>
        <p
          className="text-[13px] font-semibold undefined"
        >
          3.3999999
          %
        </p>
      </div>
      <div
        className="flex flex-col bg-medium_grey rounded p-2 gap-1 w-[125px]"
        id="token-atlas-price-card"
      >
        <p
          className="text-[9px] font-semibold text-light_grey"
        >
          1Y
        </p>
        <p
          className="text-[13px] font-semibold undefined"
        >
          6.7
          %
        </p>
      </div>
    </div>
  </div>
  <div
    className="flex flex-col gap-2 mb-20"
    id="token-atlas-info-column-stats"
  >
    <p
      className="text-[13px] font-semibold undefined"
    >
      Stats
    </p>
    <div
      className="flex flex-col rounded px-4 bg-medium_grey"
    >
      <div
        className="flex justify-between border-b border-b-dark_grey py-3"
      >
        <p
          className="text-[13px] font-normal text-light_grey undefined"
        >
          All time high
        </p>
        <p
          className="text-[13px] font-semibold undefined"
        >
          146
        </p>
      </div>
      <div
        className="flex justify-between border-b border-b-dark_grey py-3"
      >
        <p
          className="text-[13px] font-normal text-light_grey undefined"
        >
          All time low
        </p>
        <p
          className="text-[13px] font-semibold undefined"
        >
          96
        </p>
      </div>
      <div
        className="flex justify-between border-b border-b-dark_grey py-3"
      >
        <p
          className="text-[13px] font-normal text-light_grey undefined"
        >
          Total supply
        </p>
        <p
          className="text-[13px] font-semibold undefined"
        >
          300
        </p>
      </div>
      <div
        className="flex justify-between border-b border-b-dark_grey py-3"
      >
        <p
          className="text-[13px] font-normal text-light_grey undefined"
        >
          Circulating supply
        </p>
        <p
          className="text-[13px] font-semibold undefined"
        >
          170
        </p>
      </div>
      <div
        className="flex justify-between border-b border-b-dark_grey py-3"
      >
        <p
          className="text-[13px] font-normal text-light_grey undefined"
        >
          Diluted market cap
        </p>
        <p
          className="text-[13px] font-semibold undefined"
        >
          100
        </p>
      </div>
    </div>
  </div>
</div>
`;
